### PR TITLE
Linux Agent: Clean up handling of GNU 'stat' 5.3 issue

### PR DESCRIPTION
**Pull request by rawiriblundell**
The existing section for NFS and CIFS mounts individually deals with a bug in GNU 'stat', resulting in duplicated code.

This commit splits out the detection of the faulty version of 'stat' and corrects it for the entire agent, giving consistent 'stat' behaviour globally, which allows us to de-duplicate the NFS/CIFS code in the process.  This de-duplicated code is tucked away in a new function; `section_nfs()`

This commit also adds two functions: `get_cifs_mounts()` and `get_nfs_mounts()` which are used by `section_nfs()`.  These provide methods for when `/proc/mounts` exists and when it doesn't.

In other words, the handling for NFS and CIFS from the AIX agent is basically 99% merged in.

If other OSes require different approaches for getting this information, then these functions can be updated to suit.

Finally, 'sections_systemd' is renamed to 'section_systemd' in order to be consistent with other section_* functions.  This change really isn't worth a dedicated PR.

### CODE REVIEW

**Comment by rawiriblundell** [comment]
I have just tweaked this slightly to make the comment on the stat section less clunky, and to make it not-noisy with other versions of stat.  So if we happen across a non-GNU stat, whatever error message it puts out for `stat --version` will be redirected from stderr to stdout and consumed by the following `head | grep` where the `grep` will silence it (i.e. no match, so no noise)

**Comment by rawiriblundell** [comment]
Hey team, is there anything I can do to help get this moved along?

**Comment by jonaskluger** [comment]
Hello Rawiri,

thank your for your contribution!
We will have a closer look at your patch and discuss internally on how and if we can implement it into the official code base.

Best
Jonas

**Comment by jonaskluger** [comment]
Internal Ref: CMK-2769

**Comment by rawiriblundell** [comment]
Hi,
has there been any (further?) internal discussion on this PR?  It seems like a very straightforward change to me, so I'm not sure why it has been hanging around for over half a year...

Cheers

**Comment by rawiriblundell** [comment]
Continued in #163
